Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -76,27 +76,32 @@ internal class ConnectionPool {
 
     /// Initializes the pool using the provided `ConnectionString` and options.
     internal init(from connString: ConnectionString, options: ClientOptions?) throws {
+        // validate option before we bother creating pool, so we don't have to destroy it on error.
```

Comment:
I wanted to avoid calling `mongoc_client_pool_destroy` from the init. I *think* it would technically be ok because libmongoc wouldn't have started SDAM and wouldn't try to send `endSession` or anything but I don't know if we should rely on that. validating the option before pool creation seemed like the easiest way to avoid this.

Reply:
I think a more future proof way of handling this might be to wrap the whole thing in a `do { ... } catch { ... }`, where we clean up the pool and then rethrow the error in the `catch`. This way we don't really have to worry about the ordering of things. What do you think?

Reply:
I think that still suffers from the issue of having to call potentially blocking code from this initializer to do cleanup, though?

Reply:
Oh, I completely misunderstood your comment. This makes sense now. Maybe we should add a comment saying that once the pool has been created, we should not throw to avoid leaking / blocking?

Reply:
I added a bit more context to the comment here about why calling the destroy method would be undesirable